src/routes/docs/tutorials/pagination/step-7/+page.markdoc

```diff
@@ -0,0 +1,184 @@
+---
+layout: tutorial
+title: Implementing the offset pagination
+description: Discover offset pagination. The react app will access our appwrite backend using offset pagination.
+step: 7
+---
+In this step, we will discover how to add the offset pagination method to our app.
+
+# What is offset pagination? {% #what-is-offset-pagination %}
+
+Offset pagination is widely used in databases and online APIs to extract specific subsets of data from a bigger whole.It requires you to tell it where to start ("offset") and how many records to get ("limit"). In the context of software applications, offset pagination is a technique that may be employed to facilitate the navigation of content by incorporating forward and back buttons on a given page.
+
+In Appwrite, offset pagination can be achieved using the `Query.limit()` and `Query.offset()` methods. Here is an example using Appwrite's `listDocuments()` command to list the next four tasks:
+
+```js
+const page2 = await databases.listDocuments(
+    '[TODOS_DATABASE_ID]',
+    '[TODOS_COLLECTION_ID]',
+    [
+        Query.limit(4),
+        Query.offset(4)
+    ]
+);
+```
+this will run and retrieve the next four tasks starting at task-5
+
+# Implementation {% #implementation %}
```

Comment:
We don't do headings that are stacked like this.

If we have no text to go directly below implementation, you can either remove the Task... subheadings or the Implementation heading.